business/istio_validations.go

```diff
@@ -480,6 +485,102 @@ func (in *IstioValidationsService) fetchDetails(rValue *kubernetes.IstioDetails,
 	}
 }
 
+func (in *IstioValidationsService) fetchExportedResources(exportedResources *kubernetes.ExportedResources, namespace string, errChan chan error, wg *sync.WaitGroup) {
+	defer wg.Done()
+	if len(errChan) > 0 {
+		return
+	}
+	nss, err := in.businessLayer.Namespace.GetNamespaces()
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	currentVSs, err := in.fetchVirtualServices(namespace)
+	if err != nil {
+		errChan <- err
+		return
+	}
+	currentDRs, err := in.fetchDestinationRules(namespace)
+	if err != nil {
+		errChan <- err
+		return
+	}
+	currentSEs, err := in.fetchServiceEntries(namespace)
+	if err != nil {
+		errChan <- err
+		return
+	}
+	for _, ns := range nss {
+		vsList, err := in.fetchVirtualServices(ns.Name)
+		if err != nil {
+			errChan <- err
+			return
+		}
+		if containsExportToNamespacesIstioObjects(ns, currentVSs) {
+			exportedResources.VirtualServices = append(exportedResources.VirtualServices, vsList...)
+		}
+
+		drList, err := in.fetchDestinationRules(ns.Name)
+		if err != nil {
+			errChan <- err
+			return
+		}
+		if containsExportToNamespacesIstioObjects(ns, currentDRs) {
+			exportedResources.DestinationRules = append(exportedResources.DestinationRules, drList...)
+		}
+
+		seList, err := in.fetchServiceEntries(ns.Name)
+		if err != nil {
+			errChan <- err
+			return
+		}
+		if containsExportToNamespacesIstioObjects(ns, currentSEs) {
+			exportedResources.ServiceEntries = append(exportedResources.ServiceEntries, seList...)
+		}
+	}
+}
```

Comment:
For me to understand, what the `exportedResources` will hold at the end of the call?

I see that if one of the VirtualServices living in the `namespace` (currentVSs) has the `exportTo` field matching to this very same namespace (aka `containsExportToNamespacesIstioObject` returns true) then you are adding all the VirtualServices of all namespaces into the `exportedResources.VirtualServices`. Because `containsExportToNamespacesIstioObjects(ns, currentVSs)` will have always the same value in each iteration. Therefore, I'd append each `vsList` in each iteration.

So in 99% of the cases, the `exportedResources` would have all the `VirtualServices` in the cluster because 99% of the times there will be one VirtualService with no `exportedTo` field (meaning that is available to all the namespaces). And so, I'd include VirtualServices in other namespaces not exported to this namespace.